### DISCUSSION TITLE
dotnet quits unexpectedly upon Mac with SplashKit and Visual Code Studio

### DISCUSSION THREAD

**Discussion by fARTAUD** [splashkit/splashkit-core#144] [Ask a Question] [answered]
G'day from a naive ignorant student whom has no idea from downunder Australia

I am studying Object Orientated Design upon an early 2011 Mac Book Pro and for our first assignment task to check whether SplashKit installation is operating my build succed and then when command run dotnet quits unexpectedly 

Lecturer believes might be a Mac library issue and i need updated and incompatible for early 2011 Mac Book Pro? operating system to 10.15 - i have High Sierra 10.13.

To get SplashKit installed on macOS, i followed four steps:

Step 1: Install Xcode Command Line Tools
Step 2: Install SplashKit SDK
Step 3: Install Visual Studio Code
Step 4: Install Language Tools
Once you have all of these steps complete you should be setup and ready to start building SplashKit applications in the language of your choice.

Here is code needed to successfully run and attached in document with some screenshots of error messages
![Screen Shot 2021-07-17 at 2 23 19 am](https://user-images.githubusercontent.com/87668871/126209959-4b4ebb58-b5fe-4b03-a503-d113ec988160.png)

![Screen Shot 2021-07-18 at 10 18 18 am](https://user-images.githubusercontent.com/87668871/126209865-473a460c-4ebf-4e22-9d18-be4255ca89f9.png)
:-

5. Now lets try testing out that SplashKit is working correctly. Add in the extra code shown below. This will open a window, draw some text to it, display that text, and delay.
6. 
using System;
using SplashKitSDK;
public class Program {
public static void Main() {
  Console.WriteLine("Hello World");
Window w = new Window("My First Program", 200, 100); w.DrawText("Hello World", Color.Black, 10, 45); w.Refresh(60);
SplashKit.Delay(5000);

[SIT771-1.1P-TaskSheet.pdf](https://github.com/splashkit/splashkit-core/files/6843158/SIT771-1.1P-TaskSheet.pdf)

**Comment by jakerenzella** [comment, marked as answer]
Hey @fARTAUD, thanks for filing this discussion.

I agree that upgrading the operating system may be the best bet -- @macite might have some more info here. Regardless we should better list the minimum OS versions supported for MacOS.

**Reply by macite** [reply]
Yes you will need a newer version of MacOS as this version does not support the C/C++ version that we use. If you cannot update then you can still make use of the VM version. See the install instructions here: https://www.splashkit.io/articles/installation/vm/
